server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1914,13 +1849,10 @@ private void validateRootVolumeDetachAttach(VolumeVO volume, UserVmVO vm) {
         }
     }
 
-    public Volume detachVolumesFromVM(long vmId, long volumeId) {
+    public Volume detachVolumeViaDestroyVM(long vmId, long volumeId) {
```

Comment:
Why would this be specific to destroy vm ? isn't it just a detach method and didn't it already exist?